ruma-events/src/tag.rs

```diff
@@ -35,6 +36,29 @@ impl From<Tags> for TagEventContent {
     }
 }
 
+/// The name of a tag.
+#[derive(Clone, Debug, PartialEq, Eq, PartialOrd, Ord, StringEnum)]
+#[cfg_attr(not(feature = "unstable-exhaustive-types"), non_exhaustive)]
+pub enum TagName {
+    /// `m.favourite`: The user's favourite rooms. These should be shown with higher precedence
+    /// than other rooms.
+    #[ruma_enum(rename = "m.favourite")]
+    Favourite,
```

Comment:
I kinda feel like we should name this `Favorite` since I'm pretty sure we use American English everywhere else (and same for the spec). Somewhat similar to using `LowPriority` instead of `Lowpriority` below.